### ISSUE TITLE
How to fix seed value for random and nn in flair.

### ISSUE THREAD

**Issue by Dhanachandra** [flairNLP/flair#1189]
I want the flair model to output the same(loss or accuracy) for training the model multiple times. 

It was possible with the previous version i.e. 0.4.2. by adding the following piece of code in the script. 

#fix seed
`def seed_torch(seed=12345):
  random.seed(seed)
  os.environ['PYTHONHASHSEED'] = str(seed)
  np.random.seed(seed)
  torch.manual_seed(seed)
  torch.cuda.manual_seed(seed)
  torch.cuda.manual_seed_all(seed) # if you are using multi-GPU.
  torch.backends.cudnn.benchmark = False
  torch.backends.cudnn.deterministic = True

seed_torch()`

But it doesn't work with the latter versions. I tried adding this code in nn.py, samplers.py and trainer.py but it doesn't work.

**Comment by yosipk** [comment]
I haven't tried, but last two posts in [this thread](https://discuss.pytorch.org/t/random-seed-initialization/7854/17) seem related.

**Comment by Dhanachandra** [comment]
@yosipk  
Thanks for your reply. But my concern is on the flair model. Your suggested thread the same as the piece of code I mentioned above.

**Comment by yosipk** [comment]
I was referring in particular to the last message in that thread, which says that having `torch.backends.cudnn.enabled = False` and setting `num_workers = 0` is important.

I don't see that you've switched off the cuDNN in your code, so I wouldn't call it same. 

For the `num_workers = 0` part: you can set the `num_workers` in Flair trainer to 0, since it's setting the `num_workers` in PyTorch's `DataLoader` to 0, c.f. [here](https://github.com/zalandoresearch/flair/blob/master/flair/trainers/trainer.py#L261).

This will, of course, slow down the training, the thread suggests that this is a way to get the same results with the same seed.

Hope it helps!

**Comment by dhanachandran-ezdi** [comment]
@yosipk 
Thanks for the suggestion but actually it doesn't work even with your settings.

#First run:
`2019-10-10 10:23:33,557 epoch 1 - iter 0/521 - loss 0.63217676 - samples/sec: 440.24
2019-10-10 10:26:53,126 epoch 1 - iter 52/521 - loss 0.72093571 - samples/sec: 8.38`
#Second Run:
`2019-10-10 10:28:53,022 epoch 1 - iter 0/521 - loss 0.69952667 - samples/sec: 385.46
2019-10-10 10:32:26,635 epoch 1 - iter 52/521 - loss 0.71430530 - samples/sec: 7.83`
